spec/integration/data_stream/manifest.spec.yml

```
@@ -222,5 +222,19 @@ spec:
               type: array
               items:
                 type: string
+        source_mode:
+          description: |-
+            Source mode to use. This configures how the document source (`_source`) is stored
+            for this data stream.
+            If configured as `default`, this mode is not configured and it uses Elasticsearch defaults.
+            If configured as `synthetic`, it doesn't store the source, but tries to rebuild it
+            from the indexed fields when queried.
+            If no configured, users may override the setting from Fleet UI.
```

If it is configured, users can't override it?

---

I guess that if a package developer explicitly uses `source_mode: default` is because they know that the data stream doesn't work with synthetic mode, so it shouldn't be possible to override it.

For synthetic mode it may make sense to override it, as a escape hatch in case there is some unexpected problem with the data or the integration.

I can change to this if you agree.
```suggestion
            If no configured or set to `synthetic`, users may override the setting from Fleet UI.
```

---

Agree on the change. If we later find out that even when specifically set to default there is a reason to change, it would be an extension and not a breaking change.